mysql.go

```diff
@@ -307,25 +307,7 @@ func (dialector Dialector) DataTypeOf(field *schema.Field) string {
 	}
 
 	if field.AutoIncrement {
-		sqlType := "bigint"
-		switch {
-		case field.Size <= 8:
-			sqlType = "tinyint"
-		case field.Size <= 16:
-			sqlType = "smallint"
-		case field.Size <= 24:
-			sqlType = "mediumint"
-		case field.Size <= 32:
-			sqlType = "int"
-		}
-
-		if field.GORMDataType == schema.Uint {
-			sqlType += " unsigned"
-		}
-
-		sqlType += " AUTO_INCREMENT"
-
-		return sqlType
+		return string(field.DataType) + " AUTO_INCREMENT"
```

Comment:
Need to check if the data types contains `AUTO_INCREMENT` already, only add it if doesn't contains?

Reply:
if field.AutoIncrement {
return string(field.DataType) + " AUTO_INCREMENT"
}
checked?

Reply:
func (dialector Dialector) DataTypeOf(field *schema.Field) string {
	switch field.DataType {
	case schema.Bool:
		return "boolean"
	case schema.Int, schema.Uint:
		return dialector.getSchemaIntAndUnitType(field)
	case schema.Float:
		return dialector.getSchemaFloatType(field)
	case schema.String:
		return dialector.getSchemaStringType(field)
	case schema.Time:
		return dialector.getSchemaTimeType(field)
	case schema.Bytes:
		return dialector.getSchemaBytesType(field)
	}

	if field.AutoIncrement {
		return string(field.DataType) + " AUTO_INCREMENT"
	}

	return string(field.DataType)
}